bcManager/bcManager.py

```diff
@@ -1934,12 +1976,15 @@ def is_captain(self, player):
                 return True
         return False
     
-    def get_latest_account_replay_by_plat_id(self, guild, platform, plat_id):        
+    async def get_latest_account_replay_by_plat_id(self, guild, platform, plat_id):        
+        """ Get most recent replay by Platnium ID """
         bapi : ballchasing.Api = self.ballchasing_api[guild]
         data = bapi.get_replays(player_id=f"{platform}:{plat_id}", sort_by="replay-date", sort_dir="desc", count=1)
-        return next(data, {})
+        async for r in data:
+            return r # Workaround `StopAsyncIteration` exception with `__anext__()` for one replay
```

Comment:
what's going on here?

Reply:
get_replays() returns an async generator. We currently use python 3.9.x which does not actually have a neat little command for `next()` when it comes to async generators. 

I used an async for loop to safely iterate the data and just return the first result. Otherwise we hit a StopAsyncIteration exception if there is no data present using the builtin hidden `__anext__()`